sklearn/decomposition/tests/test_dict_learning.py

```diff
@@ -59,6 +59,15 @@ def test_dict_learning_reconstruction_parallel():
     code = dico.transform(X)
     assert_array_almost_equal(np.dot(code, dico.components_), X, decimal=2)
 
+def test_dict_learning_reconstruction_readonly():
+    n_components = 12
+    X.flags.writeable = False
+    dico = DictionaryLearning(n_components, transform_algorithm='lasso_cd',
+                              transform_alpha=0.001, random_state=0, n_jobs=-1)
```

Comment:
I changed this so that the test only assert that dict_learning works on memory mapped ro arrays